python/hail/keytable.py

```diff
@@ -1379,5 +1379,113 @@ def range(cls, n, num_partitions=None):
 
         return KeyTable(Env.hc(), Env.hail().keytable.KeyTable.range(Env.hc()._jhc, n, joption(num_partitions)))
 
+    @handle_py4j
+    @record_method
+    @typecheck_method(columns=oneof(strlike, listof(strlike)))
+    def ungroup(self, columns):
+        """Lifts fields of struct columns as distinct top-level columns.
+
+        **Examples**
+
+        ``kt4`` is a :py:class:`.KeyTable` with five columns: A, B, C, D and E.
+
+        >>> kt4 = hc.import_table('data/kt_example4.tsv', impute=True,
+        ...                       types={'B': TStruct(['C0', 'C1'], [TBoolean(), TString()]),
+        ...                              'D': TStruct(['cat', 'dog'], [TInt32(), TInt32()]),
+        ...                              'E': TStruct(['A', 'B'], [TInt32(), TInt32()])})
+
+        The types of each column are ``Int32``, ``Struct``, ``Boolean``, ``Struct`` and ``Struct`` respectively.
+
+        +----+--------------------------+-------+-------------------+--------------+
+        | A  |   B                      |   C   | D                 | E            |
+        +====+==========================+=======+===================+==============+
+        | 32 | {"C0":true,"C1":"hello"} | false | {"cat":5,"dog":7} | {"A":5,"B":7}|
+        +----+--------------------------+-------+-------------------+--------------+
+
+        Both B and D can be ungrouped. However, E cannot be ungrouped because the names of its
+        fields collide with existing columns in the key table.
```

Comment:
can we use name mangling here?

Reply:
I think it might be confusing to do the mangling as in the duplicate samples case. What do you think of the mangling deduplication function to be the full path (same as flatten). For example, I am worried about this case:

```
Struct{
    pValue: Float64,
    OR: Float64,
    linreg: Struct{
        pValue: Float64,
        OR: Float64
    },
    logreg: Struct{
        pValue: Float64,
        OR: Float64
    }
}
```

becoming

```
Struct{
    pValue: Float64,
    OR: Float64,
    pValue_1: Float64,
    OR_1: Float64
    pValue_2: Float64,
    OR_2: Float64
}
```

I think this is better:
```
Struct{
    pValue: Float64,
    OR: Float64,
    `linreg.pValue`: Float64,
    `linreg.OR`: Float64
    `logreg.pValue`: Float64,
    `logreg.OR`: Float64
}
```